### PR TITLE
Motion comment section list

### PR DESCRIPTION
**Pull request by FinnStutzenstein**

### CODE REVIEW

**Comment by FinnStutzenstein** [comment]
Ready to test!

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            </mat-panel-title>
+            <mat-panel-title *ngIf="editId === section.id">
+                {{ updateForm.get('name').value }}
+            </mat-panel-title>
+        </mat-expansion-panel-header>
+        <form [formGroup]="updateForm" *ngIf="editId === section.id" (keydown)="keyDownFunction($event, section)">
+            <span translate>Edit section details:</span>
+            <p>
+                <mat-form-field>
+                    <input formControlName="name" matInput placeholder="{{'Name' | translate}}">
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.html]: Added as mat-hint

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            read_groups_id: viewSection.read_groups_id,
+            write_groups_id: viewSection.write_groups_id
+        });
+    }
+
+    /**
+     * Saves the categories
+     */
+    public onSaveButton(viewSection: ViewMotionCommentSection): void {
+        if (this.updateForm.controls.name.valid) {
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: What do you mean? Are the extension panels open on reload?

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    public commentSections: ViewMotionCommentSection[] = [];
+
+    /**
+     * The current focussed formgroup
+     */
+    public updateForm: FormGroup;
+
+    public createForm: FormGroup;
+
+    public openId: number | null;
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: becuase if an autoupdate comes in, this viewModel is replaced with a fresh created. This information is lost then (you can recover it, but this is hacky). In fact that only one extension panel is open at a time, the component should take care of it.

File: client/src/app/core/services/prompt.service.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class PromptService extends OpenSlidesComponent {
+    public constructor(private dialog: MatDialog) {
+        super();
+    }
+
+    public async open(title: string, content: string): Promise<any> {
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/core/services/prompt.service.ts]: Done.

File: client/src/app/shared/components/prompt-dialog/prompt-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/shared/components/prompt-dialog/prompt-dialog.component.ts]: Done.

File: client/src/app/shared/components/prompt-dialog/prompt-dialog.component.html [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<h2 mat-dialog-title>{{ data.title | translate }}</h2>
+<mat-dialog-content>{{ data.content | translate }}</mat-dialog-content>
+<mat-dialog-actions>
+  <button mat-button [mat-dialog-close]="false" translate>Cancel</button>
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/shared/components/prompt-dialog/prompt-dialog.component.html]: Done.

File: client/src/app/site/motions/services/motion-comment-section-repository.service.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    public constructor(protected DS: DataStoreService, private dataSend: DataSendService) {
+        super(DS, MotionCommentSection);
+    }
+
+    protected createViewModel(section: MotionCommentSection): ViewMotionCommentSection {
+        return new ViewMotionCommentSection(section);
+    }
+
+    public create(section: MotionCommentSection): Observable<any> {
+        console.log('create: ', section);
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/services/motion-comment-section-repository.service.ts]: Done.

[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/services/motion-comment-section-repository.service.ts]: Done.

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        }
+    }
+
+    /**
+     * Is executed when a mat-extension-panel is opened or closed
+     * @param open true if opened, false if being closed
+     * @param viewSection the category in the panel
+     */
+    public panelClosed(viewSection: ViewMotionCommentSection): void {
+        console.log('closing', viewSection);
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: Done.

File: client/src/app/shared/components/prompt-dialog/prompt-dialog.component.html [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+  <button mat-button [mat-dialog-close]="true" translate>Yes</button>
+</mat-dialog-actions>
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/shared/components/prompt-dialog/prompt-dialog.component.html]: Done.

File: client/src/app/core/services/prompt.service.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { OpenSlidesComponent } from 'app/openslides.component';
+import { PromptDialogComponent } from '../../shared/components/prompt-dialog/prompt-dialog.component';
+import { MatDialog } from '@angular/material';
+
+/**
+ */
```
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/core/services/prompt.service.ts]: Done.

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (FinnStutzenstein) [reply client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: Ok, create button needs validation.

**Comment by FinnStutzenstein** [comment]
@tsiegleauq ping

File: openslides/motions/serializers.py [hunk 11]
```diff
@@ -314,6 +315,12 @@ class Meta:
             'read_groups',
             'write_groups',)
 
+    def create(self, validated_data):
```
Comment (normanjaeckel) [inline openslides/motions/serializers.py]: Can you please explain this change? Creating a section should trigger inform_changed_data so why do we have to call it manually again?

Comment (FinnStutzenstein) [reply openslides/motions/serializers.py]: This was strange: On creation (even with groups) I got as autoupdate always empty arrays for read and write groups. The tests we have worked fine and I looked in the DB and the groups were set there. With this, the cache gets updated, and that was not happen before. I suspect, that setting things on m2m relations (like user.groups.add(2)) does not trigger an autoupdate and not updating the cache.

If you have a workaround or a nicer place for this `inform_changed_data` please give a hint..

**Pull request merged**